extension/inet/inet_escape_functions.cpp

```diff
@@ -0,0 +1,213 @@
+
+		// the position after the end of the current match
+		start_pos = (match_pos + 1) + match.length();
+
+		if (match[0] == '#') { // true, if it represents a hexadecimal value or a Unicode point
+			int32_t num;
+			PrepareStrForTypeCasting(match);
+			if (!TryCast::Operation<string_t, int32_t>(string_t(match), num)) {
+				result += "\uFFFD";
+				continue;
+			}
+
+			if (info.invalid_charrefs.find(num) != info.invalid_charrefs.end()) {
+				auto ch = info.invalid_charrefs.at(num);
+				if (Utf8Proc::Analyze(ch.c_str(), ch.length()) == UnicodeType::INVALID) {
+					auto blob = string_t(info.invalid_charrefs.at(num));
+					auto str_blob = Blob::ToString(blob);
+					result += str_blob;
+				} else {
+					result += ch;
+				}
+
+			} else if ((0xD800 <= num && num <= 0xDFFF) || 0x10FFFF < num) {
+				result += "\uFFFD";
+			} else if (info.invalid_codepoints.find(num) != info.invalid_codepoints.end()) {
+				// skip this character
+				continue;
+			} else {
+				result += static_cast<char>(num);
+			}
+		} else { // named character reference
+			auto it = info.html5_names.find(match);
+			if (it != info.html5_names.end()) {
+				if (!AddBlob(it->second, result)) {
+					throw ConversionException("Cannot convert codepoint of %s", match);
+				}
+				continue;
+			}
+			for (idx_t x = match.length() - 1; x >= 1; --x) {
+				string substr = match.substr(0, x);
+				it = info.html5_names.find(substr);
+				if (it != info.html5_names.end()) {
+					if (!AddBlob(it->second, result)) {
+						throw ConversionException("Cannot convert codepoint of %s to utf8", match);
+					}
+					start_pos -= (match.length() - x);
+					input.set(input_data + start_pos, input_size - start_pos);
+					break;
+				}
+			}
+			if (it == info.html5_names.end()) {
+				result.append('&' + match);
+			}
+		}
+	}
+	input.AppendToString(&result);
+	return result;
+}
+
+static void EscapeInputStr(string_t &input_str, string &escaped, bool input_quote) {
+	const auto str = input_str.GetData();
+	const auto str_size = input_str.GetSize();
+	for (idx_t i = 0; i < str_size; ++i) {
+		char ch = str[i];
+		switch (ch) {
+		case '&':
+			escaped += "&amp;";
+			break;
+		case '<':
+			escaped += "&lt;";
+			break;
+		case '>':
+			escaped += "&gt;";
+			break;
+		case '"':
+			if (input_quote) {
+				escaped += "&quot;";
+			} else {
+				escaped += "\"";
+			}
+			break;
+		case '\'':
+			if (input_quote) {
+				escaped += "&#x27;";
+			} else {
+				escaped += "\'";
+			}
+			break;
+		default:
+			escaped += ch;
+		}
+	}
+}
+
+void INetFunctions::Escape(DataChunk &args, ExpressionState &state, Vector &result) {
+	auto escape_string = [&](string_t &input_str, bool input_quote) {
+		string escaped {""};
+		EscapeInputStr(input_str, escaped, input_quote);
+		return StringVector::AddString(result, escaped);
+	};
+	if (args.ColumnCount() == 1) {
+		UnaryExecutor::Execute<string_t, string_t>(args.data[0], result, args.size(),
```

Comment:
Can we first count how many bytes the result will have, folowed by `StringVector::EmptyString` and then fill up the result string with the actual escaped values? That should avoid several copies/allocations and speed things up.